web/src/settings_org.ts

```diff
@@ -358,6 +361,73 @@ function disable_create_user_groups_if_on_limited_plan(): void {
     }
 }
 
+function set_welcome_message_custom_text_visibility(): void {
+    const welcome_message_custom_text_is_configured =
+        realm.realm_welcome_message_custom_text.length > 0;
+    $("input#id_realm_enable_welcome_message_custom_text").prop(
+        "checked",
+        welcome_message_custom_text_is_configured,
+    );
+    settings_components.change_element_block_display_property(
+        "id_realm_welcome_message_custom_text",
+        welcome_message_custom_text_is_configured,
+    );
+    if (welcome_message_custom_text_is_configured) {
+        const setting_value = realm.realm_welcome_message_custom_text;
+        $("#id_realm_welcome_message_custom_text").val(setting_value);
+    } else {
+        $("#id_realm_welcome_message_custom_text").val("");
+    }
+    update_test_welcome_bot_custom_message_button_status();
+}
+
+function update_view_welcome_bot_custom_message_button_status(
+    message_id: number | undefined,
+    is_error: boolean,
+): void {
+    $("#view_welcome_bot_custom_message").remove();
+    const args = {
+        attention: is_error ? "borderless" : "quiet",
+        intent: is_error ? "danger" : "success",
+        label: is_error
+            ? $t({defaultMessage: "Error sending message"})
+            : $t({defaultMessage: "View message"}),
+        id: "view_welcome_bot_custom_message",
+    };
+
+    const view_message_button_html = render_action_button(args);
+    $("#welcome_message_custom_text_buttons_container").append($(view_message_button_html));
```

Comment:
You don't need to add the `$()` around `view_message_button_html`

Reply:
https://api.jquery.com/append/ looks like it allows just a string too; but it looks like we have this pattern a lot, so probably we should discuss in #frontend cleaning that up.